### PR TITLE
GenReconciler for kafkasource

### PR DESCRIPTION
**Pull request by lberk**
Add injected reconciler via annotation for kafkasource update codegen
Change event updates to use gen-reconciler consistent function style

### CODE REVIEW

**Comment by n3wscott** [comment]
Oh boy..... 😀😃😀😃

File: kafka/source/pkg/reconciler/controller.go [hunk 1]
```diff
@@ -57,7 +55,8 @@ func NewController(
 	deploymentInformer := deploymentinformer.Get(ctx)
 
 	c := &Reconciler{
-		Base:                reconciler.NewBase(ctx, controllerAgentName, cmw),
+		KubeClientSet:       kubeclient.Get(ctx),
```
Comment (n3wscott) [inline kafka/source/pkg/reconciler/controller.go]: Nice you got base out!!

File: kafka/source/pkg/reconciler/kafkasource.go [hunk 2]
```diff
@@ -115,25 +135,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
...
 	if _, updateStatusErr := r.updateStatus(ctx, kafka.DeepCopy()); updateStatusErr != nil {
-		logging.FromContext(ctx).Warn("Failed to update the KafkaSource", zap.Error(err))
+		pkgLogging.FromContext(ctx).Warn("Failed to update the KafkaSource", zap.Error(err))
 		r.Recorder.Eventf(kafka, corev1.EventTypeWarning, kafkaUpdateStatusFailed, "Failed to update KafkaSource's status: %v", err)
 		return updateStatusErr
 	}
 
 	return err
-}
+}*/
```
Comment (n3wscott) [inline kafka/source/pkg/reconciler/kafkasource.go]: Delete?

File: kafka/source/pkg/reconciler/kafkasource.go [hunk 3]
```diff
@@ -40,21 +37,18 @@ import (
...
-	clientset "knative.dev/eventing-contrib/kafka/source/pkg/client/clientset/versioned"
+	versioned "knative.dev/eventing-contrib/kafka/source/pkg/client/clientset/versioned"
+	reconcilerkafkasource "knative.dev/eventing-contrib/kafka/source/pkg/client/injection/reconciler/sources/v1alpha1/kafkasource"
 	listers "knative.dev/eventing-contrib/kafka/source/pkg/client/listers/sources/v1alpha1"
 	eventinglisters "knative.dev/eventing/pkg/client/listers/eventing/v1alpha1"
-	"knative.dev/eventing/pkg/reconciler"
-	"knative.dev/pkg/resolver"
-
-	"knative.dev/pkg/controller"
 	pkgLogging "knative.dev/pkg/logging"
```
Comment (n3wscott) [inline kafka/source/pkg/reconciler/kafkasource.go]: I would switch this to just be logging or no alias

File: kafka/source/pkg/reconciler/kafkasource.go (hunk 2 above)
Comment (lberk) [reply kafka/source/pkg/reconciler/kafkasource.go]: Darn, you're speedy!  Done in dc78039

File: kafka/source/pkg/reconciler/kafkasource.go (hunk 3 above)
Comment (lberk) [reply kafka/source/pkg/reconciler/kafkasource.go]: Done in 62ab377

**Comment by n3wscott** [comment]
/lgtm
/Approve

Assuming passes

**Pull request merged**